vnext/local-cli/config/configUtils.js

```diff
@@ -0,0 +1,259 @@
+    return joinedDir;
+  }
+
+  return null;
+}
+
+/**
+ * Checks if the target file path is a RNW solution file.
+ * @param {string} filePath The absolute file path to check.
+ * @return {boolean} Whether the path is to a RNW solution file.
+ */
+function isRnwSolution(filePath) {
+  return (
+    fs
+      .readFileSync(filePath)
+      .toString()
+      .search(/ReactNative/) > 0
+  );
+}
+
+/**
+ * Search for the RNW solution files under the target folder.
+ * @param {string} winFolder The absolute path to target folder.
+ * @return {array} Return the array of relative file paths.
+ */
+function findSolutionFiles(winFolder) {
+  // First search for all potential solution files
+  const allSolutions = findFiles(winFolder, '*.sln');
+
+  if (allSolutions.length === 0) {
+    // If there're no solution files, return 0
+    return [];
+  } else if (allSolutions.length === 1) {
+    // If there is exactly one solution file, assume it's it
+    return [allSolutions[0]];
+  }
+
+  var solutionFiles = [];
+
+  // Try to find any solution file that appears to be a react native solution
+  for (const solutionFile of allSolutions) {
+    if (isRnwSolution(path.join(winFolder, solutionFile))) {
+      solutionFiles.push(solutionFile);
+    }
+  }
+
+  return solutionFiles;
+}
+
+/**
+ * Checks if the target file path is a RNW lib project file.
+ * @param {string} filePath The absolute file path to check.
+ * @return {boolean} Whether the path is to a RNW lib project file.
+ */
+function isRnwDependencyProject(filePath) {
+  return (
+    fs
+      .readFileSync(filePath)
+      .toString()
+      .search(/Microsoft\.ReactNative\.Uwp\.(Cpp|CSharp)Lib\.targets/) > 0
+  );
+}
+
+/**
+ * Search for the RNW lib project files under the target folder.
+ * @param {string} winFolder The absolute path to target folder.
+ * @return {array} Return the array of relative file paths.
+ */
+function findDependencyProjectFiles(winFolder) {
+  // First, search for all potential project files
+  const allCppProj = findFiles(winFolder, '*.vcxproj');
+  const allCsProj = findFiles(winFolder, '*.csproj');
+
+  const allProjects = allCppProj.concat(allCsProj);
+
+  if (allProjects.length === 0) {
+    // If there're no project files, return 0
+    return [];
+  }
+
+  var dependencyProjectFiles = [];
+
+  // Try to find any project file that appears to be a dependency project
+  for (const projectFile of allProjects) {
+    if (isRnwDependencyProject(path.join(winFolder, projectFile))) {
+      dependencyProjectFiles.push(projectFile);
+    }
+  }
+
+  return dependencyProjectFiles;
+}
+
+/**
+ * Checks if the target file path is a RNW app project file.
+ * @param {string} filePath The absolute file path to check.
+ * @return {boolean} Whether the path is to a RNW app project file.
+ */
+function isRnwAppProject(filePath) {
+  return (
+    fs
+      .readFileSync(filePath)
+      .toString()
+      .search(/Microsoft\.ReactNative\.Uwp\.(Cpp|CSharp)App\.targets/) > 0
```

Comment:
this seems fragile.
Can we instead add a property in the project template that just tells us which one is the app project? something like
```xml
<PropertyGroup>
    <ReactNativeWindowsProjectType>App</ReactNativeWindowsProjectType>
</PropertyGroup>
```
then just read the value of this property?

Reply:
There are already two user-controlled mechanisms for specifying the right project if one can't be determined - the official `react-native.config.js` file approach and our `--proj` args to `run-windows` and `autolink-windows`. Adding another one seems overkill and confusing, especially if it means modifying the template. Plus autolinking won't work unless you're including that app target file anyway (because that target is what makes sure the generated target gets included).

If you're the .01% who is creating multiple RNW app projects in your code, or removing the import target that is necessary for making sure a RNW app builds, then I think you're responsible for also using any of the two officially supported overrides.